flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/cleanup/DefaultResourceCleaner.java

```diff
@@ -21,42 +21,64 @@
 import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.api.common.JobID;
 import org.apache.flink.runtime.concurrent.ComponentMainThreadExecutor;
+import org.apache.flink.util.concurrent.ExponentialBackoffRetryStrategy;
 import org.apache.flink.util.concurrent.FutureUtils;
+import org.apache.flink.util.concurrent.RetryStrategy;
 
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 import java.util.stream.Collectors;
 
-/** {@code DefaultResourceCleaner} is the default implementation of {@link ResourceCleaner}. */
+/**
+ * {@code DefaultResourceCleaner} is the default implementation of {@link ResourceCleaner}. It will
+ * try to clean up any resource that was added. Failure will result in an individual retry of the
+ * cleanup. The overall cleanup result succeeds after all subtasks succeeded.
+ *
+ * <p>The retry strategy is defined in {@link #DEFAULT_RETRY_STRATEGY}.
+ */
 public class DefaultResourceCleaner<T> implements ResourceCleaner {
 
+    private static final RetryStrategy DEFAULT_RETRY_STRATEGY =
```

Comment:
I wasn't sure about making this properties configurable. Initially, we said we don't need to do that as far as I remember. But thinking about it: At least the initial delay might be interesting for the user?